stripped/colabs/agregacje_kilkuletnie_stripped.py

```diff
@@ -0,0 +1,669 @@
+
+# zostawiamy kolumny 'lon', 'lat', 'pustynia' oraz kolumny, których nazwy zaczynają się od "monthly_avg"
+columns_to_leave = ['lon', 'lat', 'pustynia'] + [col for col in monthly_avg_3_july2023_an_merge.columns if col.startswith('monthly_avg')]
+monthly_avg_3_july2023_an_merge = monthly_avg_3_july2023_an_merge[columns_to_leave]
+
+monthly_avg_3_july2023_an_merge
+
+"""## Średnie 4-letnie"""
+
+pd_2023_monthly_avg_4 = pd.read_csv('/content/drive/MyDrive/BigMess/NASA/monthly_avg/4years/NASA_monthly_avg_2023.csv')
+del pd_2023_monthly_avg_4[pd_2023_monthly_avg_4.columns[0]]
+
+monthly_avg_4_july2023 = pd_2023_monthly_avg_4[pd_2023_monthly_avg_4['Month'] == 7]
+
+monthly_avg_4_july2023
+
+monthly_avg_4_july2023_an_merge = pd_2023_monthly_avg_4.merge(NASA_sample_an, left_on=['lon','lat'], right_on=['lon','lat'], how='inner')
+monthly_avg_4_july2023_an_merge
+
+"""Zostawimy tylko kolumny ze współrzędnymi, ze średnimi oraz kolumnę 'pustynia' ograniczając się jedynie do stwierdzenia czy dana lokalizacja jest pustynią czy nie."""
+
+# zostawiamy kolumny 'lon', 'lat', 'pustynia' oraz kolumny, których nazwy zaczynają się od "monthly_avg"
+columns_to_leave = ['lon', 'lat', 'pustynia'] + [col for col in monthly_avg_4_july2023_an_merge.columns if col.startswith('monthly_avg')]
+monthly_avg_4_july2023_an_merge = monthly_avg_4_july2023_an_merge[columns_to_leave]
+
+monthly_avg_4_july2023_an_merge
+
+"""## Średnie 5-letnie"""
+
+pd_2023_monthly_avg_5 = pd.read_csv('/content/drive/MyDrive/BigMess/NASA/monthly_avg/5years/NASA_monthly_avg_2023.csv')
+del pd_2023_monthly_avg_5[pd_2023_monthly_avg_5.columns[0]]
+
+monthly_avg_5_july2023 = pd_2023_monthly_avg_5[pd_2023_monthly_avg_5['Month'] == 7]
+
+monthly_avg_5_july2023
+
+monthly_avg_5_july2023_an_merge = pd_2023_monthly_avg_5.merge(NASA_sample_an, left_on=['lon','lat'], right_on=['lon','lat'], how='inner')
+monthly_avg_5_july2023_an_merge
+
+"""Zostawimy tylko kolumny ze współrzędnymi, ze średnimi oraz kolumnę 'pustynia' ograniczając się jedynie do stwierdzenia czy dana lokalizacja jest pustynią czy nie."""
+
+# zostawiamy kolumny 'lon', 'lat', 'pustynia' oraz kolumny, których nazwy zaczynają się od "monthly_avg"
+columns_to_leave = ['lon', 'lat', 'pustynia'] + [col for col in monthly_avg_5_july2023_an_merge.columns if col.startswith('monthly_avg')]
+monthly_avg_5_july2023_an_merge = monthly_avg_5_july2023_an_merge[columns_to_leave]
+
+monthly_avg_5_july2023_an_merge
+
+"""## Funkcje"""
+
+# balansowanie zbioru danych z notatnika Rafała
+class BalanceDataSet():
+  '''
+  Two techniques for handling imbalanced data.
+  '''
+  def __init__(
+      self,
+      X: DataFrame,
+      y: DataFrame
+      ) -> None:
+      self.X = X
+      self.y = y
+      assert len(self.X)==len(self.y)
+
+  def useOverSampling(
+      self,
+      randon_seed: Optional[int] = 2023
```

Comment:
Optional oznacza, że random_seed domyślnie jest `None`. Poza tym literówka w nazwie parametru.

Nie pasuje mi również umieszczenie tego random seed. Spodziewam się, że metoda init utworzy mi jak najbardziej kompletny obiekt, więc może to tam lepiej jest utworzyć pola oversample i smote

Reply:
Już poprawione